docs/source/export.rst

```diff
@@ -0,0 +1,65 @@
+
+.. note::
+
+  Modifications to :py:class:`h5py.Dataset <h5py.Dataset>` objects act *directly* on the read file on disk.
+  Changes are applied immediately and do not require exporting or writing the file.
+
+Can I export a newly instantiated container?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+No, you can only export containers that have been read from a file. The ``src_io`` argument is required in
+:py:meth:`HDMFIO.export <hdmf.backends.io.HDMFIO.export>`.
+
+Can I read a container from disk and export only part of the container?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+It depends. You can only export the root container from a file. To export the root container without certain other
+sub-containers in the hierarchy, you can remove those other containers before exporting. However, you cannot export
+only a sub-container of the container hierarchy.
+
+What happens to links when I export?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The exported file will not contain any links to the original file.
+
+All links (such as internal links (i.e., HDF5 soft links) and links to other files (i.e., HDF5 external links))
+will be preserved in the exported file.
+
+If a link to an :py:class:`h5py.Dataset <h5py.Dataset>` in another file is added to the in-memory container after
+reading it from file and then exported, then by default, the export process will create an external link to the
+existing :py:class:`h5py.Dataset <h5py.Dataset>` object. To instead copy the data from the
+:py:class:`h5py.Dataset <h5py.Dataset>` in another
+file to the exported file, pass the keyword argument ``write_args={'link_data': False}`` to
+:py:meth:`HDF5IO.export <hdmf.backends.hdf5.h5tools.HDF5IO.export>`. This is similar to passing the keyword argument
+``link_data=False`` to :py:meth:`HDF5IO.write <hdmf.backends.hdf5.h5tools.HDF5IO.write>` when writing a file with a
+copy of externally linked datasets.
+
+What happens to references when I export?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+References will be preserved in the exported file.
+NOTE: Exporting a file involves loading into memory all datasets that contain references and attributes that are
+references. The HDF5 reference IDs within an exported file may differ from the reference IDs in the original file.
+
+Can I write a newly instantiated container to two different files?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+HDMF does not allow you to write a container that was not read from a file to two different files. For example, if you
+instantiate container A and write it file 1 and then try to write it to file 2, an error will be raised. However, you
+can read container A from file 1 and then export it to file 2, with or without modifications to container A in
+memory.
```

Comment:
I would add a section on ``What happens to object_ids?`` I assume we create new object_ids on export, but it will be useful to document the behavior.

Reply:
Object IDs are actually kept the same. A Container read from a file has a particular object ID. If the read Container is exported to a new file, the exported Container maintains its original object ID. As implemented currently, the object ID is only unique within the file. We can change that behavior, but that is how it stands currently.

It would be useful to have an ID that is unique on every write. In NWB, I suggest we use the `identifier` field for this.

Reply:
Thanks for the clarification. It would be good to at least document this behavior. Ideally, I think this should be a parameter on the export function to ``keep_object_ids=True``. I think it is fine to address this as a separate issue (rather than delaying the merge of this PR) if adding the option to generate new object_ids is tricky.